experimental/kernels/reduce.cpp

```diff
@@ -0,0 +1,420 @@
+#include "gpu.hpp"
+#include <array>
+#include <cstdio>
+#include <future>
+#include <random>
+#include <algorithm>
+#include "utils/array_utils.hpp"    // show, isclose, randn, randint
+#include "kernels.h"
+
+using namespace gpu;
+
+#define LIMITS { \
```

Comment:
this is fine as is for the experimental implementation, i wonder if there's a way we can populate this by polling system properties?

Reply:
https://github.com/AnswerDotAI/gpu.cpp/issues/72